### ISSUE TITLE
Youtube set thumbnail, onUploadProgress doesn't work

### ISSUE THREAD

**Issue by kdong007** [googleapis/google-api-nodejs-client#1820]
#### Environment details

  - OS: mac
  - Node.js version: v10.15.3
  - npm version: yarn 1.15.2
  - `googleapis` version: 42.0.0

#### Steps to reproduce

```js
getYoutube().thumbnails.set({
  videoId: id,
  media: {
    mimeType: "image/jpeg",
    body: thumbnailStream,
  },
}, {
  onUploadProgress: e => console.log("onUploadProgress", e),
});
```

callback not fired

**Comment by bcoe** [comment]
@JustinBeckwith do you remember the reasoning behind only performing multi-part uploads if the discovery document has [request.$ref](https://github.com/googleapis/google-api-nodejs-client/blob/master/src/generator/templates/api-endpoint.njk#L152).

It looks like this API endpoint `youtube.thumbnails.set` might potentially support multipart uploading:

```json
      "protocols": {
       "simple": {
        "multipart": true,
        "path": "/upload/youtube/v3/thumbnails/set"
       },
       "resumable": {
        "multipart": true,
        "path": "/resumable/upload/youtube/v3/thumbnails/set"
       }
      }
```

But unlike most methods, it has no `request` parameter described, and in turn doesn't expose `requestBody`, which is used by `apirequest.ts` to decide between `multipart` vs., `media` upload.

**Comment by JustinBeckwith** [comment]
I think these are unrelated.  The code you linked to only applies to the `requestBody` parameter.  For better or for worse, if there's a media upload this code path goes through the `{ media: { body: ... } ` code path.  That *does* provide multipart upload support, and should be triggering that `onUploadProgress` method.  

There's a really good example of using these together here:
https://github.com/googleapis/google-api-nodejs-client/blob/master/samples/youtube/upload.js

I would take that code, and try to re-create the issue specifically with `youtube.thumbnails.set` :)

**Comment by bcoe** [comment]
hey @kdong007 I've released `googleapis@48.0.0`, which should allow you to get `uploadProgress` for thumbnails. You will need to add an additional parameter: `requestBody: {}` (which will provide a hint that a multipart POST should be used).

I've added a unit test you can reference here:

https://github.com/googleapis/google-api-nodejs-client/blob/master/test/test.media.ts#L146

I've performed end to end testing with the youtube API, and was able to get the progress events 👌

**Comment by kdong007** [comment]
@bcoe thanks for making this works.
